provider/core/src/datagen/iter.rs

```diff
@@ -5,37 +5,32 @@
 //! Collection of iteration APIs for data providers.
 
 use crate::prelude::*;
+use icu_locid::Locale;
 
-/// A [`DynProvider`] that can iterate over all supported [`ResourceOptions`] for a certain key.
+/// A [`DynProvider`] that can iterate over all supported [`Locale`]s for a certain key.
```

Comment:
Issue: I am not aligned on the motivation for this change (trying to remove ResourceOptions from public APIs).